src/plot_api/plot_api.js

```diff
@@ -2954,6 +2954,11 @@ function getDiffFlags(oldContainer, newContainer, outerparts, opts) {
             flags.rangesAltered[outerparts[0]] = 1;
         }
 
+        // clear _inputDomain on cartesian axes with altered domains
+        if(AX_DOMAIN_RE.test(astr)) {
+            nestedProperty(newContainer, '_inputDomain').set(null);
```

Comment:
This piece is the `react` equivalent to this block 

https://github.com/plotly/plotly.js/blob/323195eb9fef0bf527742d5b369d8f22c00dfd7d/src/plot_api/plot_api.js#L2133-L2135

called during `relayout`.